Code/GraphMol/RascalMCES/RascalResult.cpp

```diff
@@ -0,0 +1,817 @@
+    }
+  }
+  molFrags->commitBatchEdit();
+  return molFrags;
+}
+
+// Calculate a score for how many bonds in the clique don't match
+// cyclic/non-cyclic
+int RascalResult::calcRingNonRingScore() const {
+  if (!d_mol1 || !d_mol2) {
+    return 0;
+  }
+
+  int score = 0;
+  for (const auto &bm : d_bondMatches) {
+    auto nbr1 = d_mol1->getRingInfo()->numBondRings(bm.first);
+    auto nbr2 = d_mol2->getRingInfo()->numBondRings(bm.second);
+
+    if ((nbr1 && !nbr2) || (!nbr1 && nbr2)) {
+      ++score;
+    }
+  }
+  return score;
+}
+
+// Calculate a score for how well the atoms in the clique from mol1 match the
+// atoms for the clique in mol2.  The atom scores are made up of H count and
+// summed for the molecule. Its so that, for example, an OH in mol1 that could
+// match an OH or OMe matches the OH for preference.
+int RascalResult::calcAtomMatchScore() const {
+  if (!d_mol1 || !d_mol2) {
+    return 0;
+  }
+  int score = 0;
+  for (const auto &am : d_atomMatches) {
+    int num_h_1 = d_mol1->getAtomWithIdx(am.first)->getTotalNumHs();
+    int num_h_2 = d_mol2->getAtomWithIdx(am.second)->getTotalNumHs();
+    score += std::abs(num_h_1 - num_h_2);
+  }
+  return score;
+}
+
+int RascalResult::calcMaxDeltaAtomAtomDistScore() const {
+  // Possibly this could be improved, to be the total of the minimum distances
+  // between each fragment.
+  if (d_atomMatches.empty()) {
+    return 0;
+  }
+  // These arrays are cached so shouldn't be deleted.  The final 'true' in the
+  // call is to force recalculation, just in case there's some other type copied
+  // over from the input molecule.
+  const auto *mol1Dists =
+      RDKit::MolOps::getDistanceMat(*d_mol1, false, false, true);
+  const auto *mol2Dists =
+      RDKit::MolOps::getDistanceMat(*d_mol2, false, false, true);
+
+  int score = 0;
+  auto dist = [](int idx1, int idx2, const double *dists,
+                 int num_atoms) -> int {
+    return int(std::nearbyint(dists[idx1 * num_atoms + idx2]));
+  };
+  for (size_t i = 0; i < d_atomMatches.size() - 1; ++i) {
+    for (size_t j = i + 1; j < d_atomMatches.size(); ++j) {
+      auto d1 = dist(d_atomMatches[i].first, d_atomMatches[j].first, mol1Dists,
+                     d_mol1->getNumAtoms());
+      auto d2 = dist(d_atomMatches[i].second, d_atomMatches[j].second,
+                     mol2Dists, d_mol2->getNumAtoms());
+      auto deltaDist = abs(d1 - d2);
+      if (deltaDist > score) {
+        score = deltaDist;
+      }
+    }
+  }
+  return score;
+}
+
+int RascalResult::calcLargestFragSize() const {
+  if (!d_mol1 || !d_mol2) {
+    return 0;
+  }
+  std::unique_ptr<RDKit::ROMol> mol1_frags(makeMolFrags(1));
+  std::vector<int> mapping;
+  auto numFrags = RDKit::MolOps::getMolFrags(*mol1_frags, mapping);
+  int lfs = -1;
+  for (unsigned int i = 0; i < numFrags; ++i) {
+    auto fragSize = std::count(mapping.begin(), mapping.end(), i);
+    if (fragSize > lfs) {
+      lfs = fragSize;
+    }
```

Comment:
Could be
```suggestion
    lfs = std::max(lfs,fragSize);
```

Reply:
As it stood, clang objected to that, because auto makes fragSize a long, and ifs is explicitly declared as int.  I have re-factored it so that auto can be used for both, and it's marginally more efficient a well.